### ISSUE TITLE
Display error after "Delete all messages"

### ISSUE THREAD

**Issue by qinyeli** [psturmfels/ios-team#21]
Once "Delete all messages", the next message would  not be displayed properly. (Seems that the the new message and the deleted messages shows in the same table cell.)

**Comment by qinyeli** [comment]
A guess of the cause of the problem: reusable cell.

```
let reusableCell: AnyObject = tableView.dequeueReusableCell(withIdentifier: "AnonymouseTableViewCell", for: indexPath)
```

When we delete all the message, we call `dataController.clearContext()` in `settingViewController`.
At this time, dequeReusableCell would think that there is no cell.
so the next time we try to insert into tableView, dequeReusableCell would serve `reusableCell` with the same cell it just used, thinking that it is empty.

**Comment by qinyeli** [comment]
Further observation:
* Persistent storage does not work?

**Comment by psturmfels** [comment]
Yeah, I don't know how to solve this problem. Also, please leave comments in appropriate issues.

**Comment by psturmfels** [comment]
Fixed with 479557852aa6fc4d0585707c8ad198b8bc84bf13. The issue was with correctly implementing `prepareForReuse`. This should also have fixed the strange UI issues that we had been encountering.
